errai-codegen/src/main/java/org/jboss/errai/codegen/meta/impl/apt/APTAnnotation.java

```diff
@@ -115,63 +110,31 @@ private static Object convertValue(final Object value, final Class<?> arrayType)
   private static Object convertToArrayValue(final Object value, final Class<?> arrayTypeHint) {
     return ((List<?>) value).stream()
             .map(av -> ((AnnotationValue) av).getValue())
-            .map(v -> convertValue(v, null))
+            .map(v -> convertValue(v, arrayTypeHint))
             .toArray(n -> buildArray(arrayTypeHint, n));
   }
 
-  private static Object[] buildArray(final Class<?> arrayTypeHint, int n) {
+  private static Object[] buildArray(final Class<?> arrayTypeHint, final int n) {
     if (arrayTypeHint.equals(Class[].class)) {
       return (Object[]) Array.newInstance(MetaClass.class, n);
     } else if (arrayTypeHint.equals(Annotation[].class)) {
       return (Object[]) Array.newInstance(MetaAnnotation.class, n);
+    } else if (arrayTypeHint.equals(Enum[].class)) {
+      return (Object[]) Array.newInstance(MetaEnum.class, n);
     } else {
       return (Object[]) Array.newInstance(arrayTypeHint.getComponentType(), n);
     }
   }
 
-  @Deprecated
-  private static Class<?> unsafeLoadClass(final TypeMirror value) {
-    switch (value.getKind()) {
-    case ARRAY: {
-      TypeMirror cur = value;
-      int dim = 0;
-      do {
-        cur = ((ArrayType) cur).getComponentType();
-        dim += 1;
-      } while (cur.getKind().equals(TypeKind.ARRAY));
-      final Class<?> componentClazz = unsafeLoadClass(cur);
-      final int[] dims = new int[dim];
-      final Object array = Array.newInstance(componentClazz, dims);
-
-      return array.getClass();
-    }
-    case DECLARED:
-      final String fqcn = ((TypeElement) ((DeclaredType) value).asElement()).getQualifiedName().toString();
-      try {
-        return Class.forName(fqcn);
-      } catch (final ClassNotFoundException e) {
-        throw new IllegalArgumentException(format("Cannot load class object for [%s].", fqcn));
-      }
-    case BOOLEAN:
-      return boolean.class;
-    case BYTE:
-      return byte.class;
-    case CHAR:
-      return char.class;
-    case DOUBLE:
-      return double.class;
-    case FLOAT:
-      return float.class;
-    case INT:
-      return int.class;
-    case LONG:
-      return long.class;
-    case SHORT:
-      return short.class;
-    case VOID:
-      return void.class;
-    default:
-      return throwUnsupportedTypeError(value);
-    }
+  @Override
+  public Map<String, Object> values() {
+    return values.entrySet()
+            .stream()
+            .filter(e -> !isNonbinding(e.getKey()))
```

Comment:
Same as before: the logic of filtering out NonBinding attributes probably should be external to this implementation.

Reply:
Same as above.